### PR TITLE
Multivariate factorization

### PR DESCRIPTION
**Pull request by iliailmer**
Trying to address #42

### CODE REVIEW

File: src/util.jl [hunk 1]
```diff
@@ -206,7 +206,10 @@ function fast_factor(poly::MPolyElem{fmpq})
     prelim_factors = uncertain_factorization(poly)
     cert_factors = map(pair -> pair[1], filter(f -> f[2], prelim_factors))
     uncert_factors = map(pair -> pair[1], filter(f -> !f[2], prelim_factors))
-    append!(cert_factors, factor_via_singular(uncert_factors))
+    for p in uncert_factors
+        factors = Nemo.factor(p)
+        append!(cert_factors, factors[1])
```
Comment (pogudingleb) [inline src/util.jl:211]: I am not sure `factors[1]` is well-defined (this is also why the tests fail). `factor(p)` returns an iterator of type `Fac`, so I would suggest something like `[x[1] for x in Nemo.factor(p)]`

Comment (iliailmer) [reply src/util.jl:211]: Yep, just realized that the error is from the indexing `[1]` 😄 .

**Comment by pogudingleb** [comment]
Thank you, @iliailmer !
Could you remove `factor_with_singular` since it is not longer used?

**Comment by iliailmer** [comment]
@pogudingleb sorry, forgot to rename commit, but the function is removed

**Pull request merged**